packages/mds-db/index.ts

```diff
@@ -290,12 +290,16 @@ async function readDeviceIds(provider_id?: UUID, skip?: number, take?: number):
 }
 
 // TODO: FIX updateDevice/readDevice circular reference
-async function readDevice(device_id: UUID): Promise<Recorded<Device>> {
+async function readDevice(device_id: UUID, provider_id?: UUID): Promise<Recorded<Device>> {
   return new Promise((resolve, reject) => {
     // read from pg
     getReadOnlyClient().then(client => {
-      const sql = `SELECT * FROM ${schema.DEVICES_TABLE} WHERE device_id=$1`
+      let sql = `SELECT * FROM ${schema.DEVICES_TABLE} WHERE device_id=$1`
       const values = [device_id]
+      if (provider_id) {
+        sql += ` AND provider_id=$2`
```

Comment:
This could be done as follows: ``const sql = `SELECT * FROM ${schema.DEVICES_TABLE} WHERE device_id=${vals.add(device_id)}${provider_id ? ` AND provider_id=${vals.add(provider_id)}` : ''}``

Reply:
That's what I said! Max replies: 'Sure, and we could do everything in one line of APL too!'

Reply:
@karcass Knows APL? I took a whole semester of it :-)

Reply:
It's not so much that I dislike multiple lines; I just dislike `let`.